addons/account/models/account_move.py

```diff
@@ -3690,6 +3696,26 @@ def _get_fields_onchange_balance_model(self, quantity, discount, amount_currency
             vals = {'price_unit': 0.0}
         return vals
 
+    @api.model
+    def _get_tax_exigible_domain(self):
+        """ Returns a domain to be used to identify the move lines that are allowed
+        to be taken into account in the tax report.
+        """
+        return [
+            # Lines on moves without any payable or receivable line are always exigible
+            '|', ('move_id.always_tax_exigible', '=', True),
+
+            # Lines with only tags are always exigible
+            '|', '&', ('tax_line_id', '=', False), ('tax_ids', '=', False),
+
+            # Lines from CABA entries are always exigible
+            '|', ('move_id.tax_cash_basis_rec_id', '!=', False),
+
+            # Lines from non-CABA taxes are always exigible
+            '|', ('tax_line_id.tax_exigibility', '!=', 'on_payment'),
+            ('tax_ids.tax_exigibility', '!=', 'on_payment'), # So: exigible if at least one tax from tax_ids isn't on_payment
+        ]
```

Comment:
I'm not saying this doesn't work, but the tax report still has to deal with the base lines being duplicated for cash basis taxes mixed with another tax non-cash basis. 

So, why no keeping the boolean on the line instead of moving it on the move?